sympy/solvers/bivariate.py

```diff
@@ -196,6 +198,50 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper routine to replace ``t`` with ``+/- symbol`` in ``expr`` to
+        return a list of solutions from the two ``expr`` generated.
+
+        For example, if ``expr = 2*log(t) + x/2` then solutions for
+        ``2*log(x) + x/2 = 0`` and ``2*log(-x) + x/2 = 0`` would be returned.
+
+        For an expression like ``eq = x**2*g(x) = 1``, if we take the log of
+        both sides we obtain ``log(x**2) + log(g(x)) = 0``. If x is positive
+        then this simplifies to ``2*log(x) + log(g(x)) = 0``;
+        the Lambert-solving routines will return solutions for this, but we
+        must also consider the solutions for  ``2*log(-x) + log(g(x))`` since
+        those must also be a solution of ``eq`` which has the same value
+        whether the ``x`` in ``x**2`` is negated. If `g(x)` does not have even
+        powers of symbol then we don't want to replace the ``x`` there
+        with ``-x``. So the role of the ``t`` in the expression received by
+        this function is to mark where ``+/-x`` should be inserted before
+        obtaining the Lambert solutions.
+
+        Parameters
```

Comment:
If you are going to keep this (from Parameters down) it should be better incorparated with the above since they are both saying the same thing.

Reply:
Okay

Reply:
I placed them above. Should I remove the explanation also from there?